src/autodiff_nlsmodel.jl

```diff
@@ -11,22 +12,37 @@ type ADNLSModel <: AbstractNLSModel
 
   # Function
   F :: Function
+  c :: Function
 end
 
 function ADNLSModel(F :: Function, x0 :: Vector, m :: Int;
                     name :: String = "Generic",
                     lvar :: Vector = fill(-Inf, length(x0)),
-                    uvar :: Vector = fill(Inf, length(x0))
+                    uvar :: Vector = fill(Inf, length(x0)),
+                    y0 :: Vector = Float64[],
+                    c :: Function = (args...)->throw(NotImplementedError("cons")),
+                    lcon :: Vector = Float64[],
+                    ucon :: Vector = Float64[]
                    )
   nvar = length(x0)
+  ncon = maximum([length(lcon); length(ucon); length(y0)])
```

Comment:
I think it should be an error if they were not of the same length. It's too confusing.

Reply:
I'll forbid empty lcon and ucon then, and default y0 to zeros.

Reply:
Ok.